doc/tutorials/changetracking.md

```diff
@@ -0,0 +1,112 @@
+# Changetracking
+
+Elektra provides developers of applications and plugins a way to determine changes made to a `KeySet` relative to the last known state of the key database.
+This can be useful if you are writing a plugin that is only supposed to working with changed data.
+
+## Basics
+
+The two headers you have to use are `kdbchangetracking.h` and `kdbdiff.h`.
+Those declare the `elektraChangeTracking*` and `elektraDiff*` methods.
+
+The two main data structures you will encounter are `ChangeTrackingContext` and `ElektraDiff`.
+
+## Getting the difference between KeySets
+
+If all you want to do is get the difference between two in-memory `KeySet` objects, use the function `elektraDiffCalculate`.
+
+```c
+KeySet * originalKeySet;
+KeySet * modifiedKeySet;
+Key * parentKey;
+
+// Calculate the difference of the keys below and including parentKey
+ElektraDiff * diff = elektraDiffCalculate (modifiedKeySet, originalKeySet, parentKey);
+
+// Extract useful information, see section 'Working with the diff'
+
+elektraDiffDel (diff);
+```
+
+## Getting the changes within a transaction in a plugin
+
+If you are writing a plugin, you can use `elektraChangeTrackingGetContextFromPlugin` to get the current `ChangeTrackingContext`.
+Then, use `elektraChangeTrackingCalculateDiff` to calculate the changes to the KDB.
+
+```c
+int myPluginSet (Plugin * handle, KeySet * returned, Key * parentKey)
+{
+	ChangeTrackingContext * ctx = elektraChangeTrackingGetContextFromPlugin (handle);
+	ElektraDiff * diff = elektraChangeTrackingCalculateDiff (returned, ctx, parentKey);
+
+	// Extract useful information, see section 'Working with the diff'
+
+	elektraDiffDel (diff);
+	return 1;
+}
+```
+
+## Getting the changes as an application developer
+
+If you are elektrifying your app, you can use `elektraChangeTrackingGetContextFromKdb` to get the current `ChangeTrackingContext` for your KDB instance.
+Then, use `elektraChangeTrackingCalculateDiff` to calculate the changes to the KDB.
+
+```c
+KDB * kdb;
+
+ChangeTrackingContext * ctx = elektraChangeTrackingGetContextFromPlugin (kdb);
```

Comment:
I assume you meant:

```suggestion
ChangeTrackingContext * ctx = elektraChangeTrackingGetContextFromKdb (kdb);
```

Please create an example in examples/ and see if it really compiles.